### PR TITLE
[DAF-4166]

### PR DESCRIPTION
**Pull request by ghost**
PIP displays a black screen when losing connection to the network for too long so I need an event to listen to when the player displays a black screen
https://dw-ml-nfc.atlassian.net/browse/DAF-4166

### CODE REVIEW

File: lib/src/configuration/better_player_event_type.dart [hunk 1]
```diff
@@ -32,5 +32,6 @@ enum BetterPlayerEventType {
   tapPauseButtonInPIP, // Android only. Fire when tap pause button on PIP window.
   tapExternalPlayButton, // Android only. Fire when tap play button from outside the app (e.g. PIP, Notification).
   tapExternalPauseButton, // Android only. Fire when tap pause button from outside the app (e.g. PIP, Notification).
-  finishedPlayInLooping, // Ios only. Trigger when postion = duration in looping mode
+  finishedPlayInLooping, // Ios only. Trigger when postion = duration in looping mode,
+  failedToPlayToEndTime, // Ios only. Trigger when player display black screen
```
Comment (mvn-thanhluu-hn) [inline lib/src/configuration/better_player_event_type.dart]:
```suggestion
  failedToPlayToEndTime, // Ios only. Trigger when player cannot play to the end (black screen in PIP, ...)
```

**Review by mvn-thanhluu-hn — APPROVED** [review]
LGTM.

Please check my suggestion 🙏

**Review by noboru-i — APPROVED** [review]
Sorry.
I cannot understand how we use `failedToPlayToEndTime` event.
But the code seems ok.
LGTM.

**Comment by HiroshiOshiro** [comment]
Just in case, FYI

I think one of the reason of this issue in LIVE is [there is reloading](https://github.com/dwango-nfc/dwango-app-ch/blob/3024475613231fa606f4dfa2beb412fefad30516/lib/presentation/pages/player/player_view.dart#L863) when Websocket disconnect. I assume in iOS, the websocket disconnection happen while after internet disconnection.

To solve this issue, I think need to avoid reload at websocket disconeection in PIP and background mode, instead of that, do `_listenToNetworkStatusChange()`? (not so sure)

**Comment by ghost** [comment]
@noboru-i  In pip mode, if we turn off the network, after about 1-2 minutes, the pip window will display a black screen instead of the last frame. This seems to be the default behavior of the iOS. I need to listen to the event when the black screen appears.

**Pull request merged**